code/datums/spawners_menu.dm

```diff
@@ -38,8 +38,13 @@
 	if(..())
 		return
 
-	var/spawner_ref = pick(GLOB.mob_spawners[params["name"]])
+	var/list/L = GLOB.mob_spawners[[params["name"]]]
+	if(!L || !L.len)
+		return
+	var/spawner_ref = pick(L)
 	var/obj/effect/mob_spawn/MS = locate(spawner_ref) in GLOB.poi_list
+	if(!MS)

```

Comment:
This is the only check that is really needed, the others aren't needed.

Reply:
Pick from an empty list is still a runtime.